Yosemite/Yosemite/Actions/SystemPluginAction.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+/// Defines all actions supported by `SystemPluginStore`
+///
+public enum SystemPluginAction: Action {
+
+    /// Synchronize all system plugins for a site given its ID
+    ///
+    case synchronizeSystemPlugins(siteID: Int64, onCompletion: (Result<Void, Error>) -> Void)
+
+    /// Fetch all system plugins for a site given its ID
+    ///
+    case fetchSystemPlugins(siteID: Int64, onCompletion: ([SystemPlugin]?) -> Void)
```

Comment:
Could you explain the use case for this action? I don't think we need it 🤔

Reply:
Hi! As we talked about, in the tool that we will build to know if a feature could be enabled by his plugins, we will need something to get all plugins. So I think we will need a method to retrieve the plugins saved in the storage layer.

Reply:
Sure, but why we would need to fetch all of them? I think fetching just one should be enough, don't you think?

Reply:
Yes, If we do the tools in that way, of course. In the end, I think it depends on if you want to load all plugins and check if a feature is enabled with that list, or if you want to search a specific plugin to know if a feature should be enabled or not. I will change with the second option.